src/osd/OSD.cc

```diff
@@ -7332,13 +7330,9 @@ void OSD::ms_fast_dispatch(Message *m)
     tracepoint(osd, ms_fast_dispatch, reqid.name._type,
         reqid.name._num, reqid.tid, reqid.inc);
   }
-#ifdef HAVE_JAEGER
-  op->set_osd_parent_span(dispatch_span);
-  if (op->osd_parent_span) {
-    auto op_req_span = jaeger_tracing::child_span("op-request-created", op->osd_parent_span);
-    op->set_osd_parent_span(op_req_span);
-  }
-#endif
+  auto op_req_span = tracer.start_span("op-request-created", dispatch_span);
+  std::swap(op->osd_parent_span, op_req_span);
```

Comment:
it doesn't look like we need this extra `op_req_span` variable, this could just be
```c++
op->osd_parent_span = tracer.start_span("op-request-created", dispatch_span);
```